docs/configuration.md

```diff
@@ -103,6 +103,7 @@ Unless you need to set a non-default value, it is recommended to only populate o
 | `TX_DEFAULT_GAS`                            | "400000"                              | Default gas for transactions that do not specify gas.                                                                                                                                                                                                                                                                                                                                                                                                                                      |
 | `USE_ASYNC_TX_PROCESSING`                   | "true"                                | Set to `true` to enable `eth_sendRawTransaction` to return the transaction hash immediately after passing all prechecks, while processing the transaction asynchronously in the background.                                                                                                                                                                                                                                                                                                |
 | `MAX_TRANSACTION_FEE_THRESHOLD`             | "15000000"                            | Used to set the max transaction fee. This is the HAPI fee which is paid by the relay operator account.                                                                                                                                                                                                                                                                                                                                                                                     |
+| `MAX_GAS_ALLOWANCE_HBAR`                    | "0"                            | The maximum amount, in hbars, that the relay is willing to pay to complete the transaction in case the senders don't provide enough funds to pay for their transactions. Please note, in case of full subsidized transactions, the sender must set the gas price to `0` and the relay must configure both the `GAS_PRICE_TINY_BAR_BUFFER`, to a number big enough to cover the discrepancy between the base gas price and `0`, and the `MAX_GAS_ALLOWANCE_HBAR` parameters, with the max amount of hbars the relay is willing to spend to execute the transaction.      |
```

Comment:
I'm not sure about setting `GAS_PRICE_TINY_BAR_BUFFER` in a fully subsidized JSON-RPC Relay. It seems `GAS_PRICE_TINY_BAR_BUFFER` has a different purpose. We might need to change the precheck to account for the `MAX_GAS_ALLOWANCE_HBAR` variable.

Reply:
The `MAX_GAS_ALLOWANCE_HBAR` is set by the operator, so I'm not sure it needs a dedicated pre-check.

About `GAS_PRICE_TINY_BAR_BUFFER`, you are right, that serve a different purpose, and that's fine.

That parameter is mentioned here only as a reference, because in case the user set gasPrice=0, a low buffer value (that is probably what you want in a classic JSON-RPC relay configuration) will block those transactions during the pre-check.

So, in case an operator want to support gasPrice=0 transactions, it must update the `GAS_PRICE_TINY_BAR_BUFFER` parameter also accordingly.

Reply:
> So, in case an operator want to support gasPrice=0 transactions, it must update the GAS_PRICE_TINY_BAR_BUFFER parameter also accordingly.

Exactly. I believe this is not the best UX for operators, as we are requiring them to update the unrelated `GAS_PRICE_TINY_BAR_BUFFER` variable.

Reply:
We can think about improving the operators UX in general, but probably just providing a preset of config values for the different use cases they want to achieve is enough.

The `GAS_PRICE_TINY_BAR_BUFFER` is mentioned in the comments because it's involved in the process, but it's not a constraint to set it to a specific value. Operators can still decide to keep that value small as they prefer, and pay only for txs which costs is in that specific cost range.

BTW, the scope of this PR is very limited: it's just enable operators setting the `MAX_GAS_ALLOWANCE_HBAR` value as they prefer, like they do for other parameters, so I would suggest to postpone further improvements for another PR.

Reply:
I modified the comments to adhere to the new precheck behaviour.